ddtrace/settings/config.py

```diff
@@ -607,16 +627,72 @@ def __setattr__(self, key, value):
         if key == "_config":
             return super(self.__class__, self).__setattr__(key, value)
         elif key in self._config:
-            self._config[key].set_code(value)
-            self._notify_subscribers([key])
+            self._set_config_items([(key, value, "code")])
             return None
         else:
             return super(self.__class__, self).__setattr__(key, value)
 
+    def _set_config_items(self, items):
+        # type: (List[Tuple[str, Any, _ConfigSource]]) -> None
+        for key, value, origin in items:
+            self._config[key].set_value_source(value, origin)
+
+        from ..internal.telemetry import telemetry_writer
```

Comment:
The telemetry writer needs to be initialized after the global config object. Now updating the config relies on the global telemetry being initialized. We will need to refactor this circular dependency. I don't think issue should block this PR but it needs to be solved before we migrate the DD_SERVICE, DD_ENV, DD_VERSION, and DD_INSTRUMENTATION_TELEMETRY_ENABLED.

Reply:
Yep agreed.